keycloak-openapi-transformer/src/components/schemas.rs

```diff
@@ -32,58 +33,71 @@ pub fn parse_schemas(
-            .split(", ")
-            .map(std::string::ToString::to_string)
-            .collect();
-        Some(openapiv3::Type::String(openapiv3::StringType {
-            enumeration: enumerations,
-            ..Default::default()
-        }))
-    } else {
-        None
-    }
+fn list_of_type(raw_type: &str) -> Option<openapiv3::Type> {
+    const LIST_PREFIX: &str = "List  of ";
+
+    let inner_type = raw_type.strip_prefix(LIST_PREFIX)?;
+
+    Some(openapiv3::Type::Array(openapiv3::ArrayType {
+        items: Some(parse_type_boxed(inner_type)),
+        min_items: None,
+        max_items: None,
+        unique_items: false,
+    }))
 }
 
-fn array_type(raw_type: &str) -> Option<openapiv3::Type> {
-    const START: &str = "< ";
-    const END: &str = " > array";
-    if raw_type.starts_with(START) && raw_type.ends_with(END) {
-        let inner_type = raw_type.get(START.len()..raw_type.len() - END.len())?;
-        Some(openapiv3::Type::Array(openapiv3::ArrayType {
-            items: parse_type_boxed(inner_type),
-            min_items: None,
-            max_items: None,
-            unique_items: false,
-        }))
-    } else {
-        None
-    }
+fn set_of_type(raw_type: &str) -> Option<openapiv3::Type> {
+    const SET_PREFIX: &str = "Set  of ";
+
+    let inner_type = raw_type.strip_prefix(SET_PREFIX)?;
+
+    Some(openapiv3::Type::Array(openapiv3::ArrayType {
+        items: Some(parse_type_boxed(inner_type)),
+        min_items: None,
+        max_items: None,
+        unique_items: true,
+    }))
 }
 
-fn csv_array(raw_type: &str) -> Option<openapiv3::Type> {
-    const START: &str = "< ";
-    const END: &str = " > array(csv)";
-    if raw_type.starts_with(START) && raw_type.ends_with(END) {
-        let inner_type = raw_type.get(START.len()..raw_type.len() - END.len())?;
-        Some(openapiv3::Type::Array(openapiv3::ArrayType {
-            items: parse_type_boxed(inner_type),
-            min_items: None,
-            max_items: None,
-            unique_items: false,
-        }))
-    } else {
-        None
+fn map_of_type(raw_type: &str) -> Option<openapiv3::Type> {
+    const MAP_PREFIX: &str = "Map  of ";
+
+    let inner_type_str = raw_type.strip_prefix(MAP_PREFIX)?;
+    let inner_type = parse_type(inner_type_str);
+
+    let map = openapiv3::Type::Object(ObjectType {
+        additional_properties: Some(openapiv3::AdditionalProperties::Schema(Box::new(
+            inner_type,
+        ))),
+        ..Default::default()
+    });
+
+    Some(map)
+}
+
+fn map_type(raw_type: &str) -> Option<openapiv3::Type> {
+    const MAP_PREFIX: &str = "Map[";
+
+    if !raw_type.starts_with(MAP_PREFIX) {
+        return None;
     }
+
+    let inner_type_str = raw_type.strip_prefix(MAP_PREFIX)?.strip_suffix(']')?;
+
+    let inner_type = parse_type(inner_type_str);
+
+    let map = openapiv3::Type::Object(ObjectType {
+        additional_properties: Some(openapiv3::AdditionalProperties::Schema(Box::new(
```

Comment:
Also not sure about this structure. It was hard to tell if this is the intended format. I would need to compare with the actual responses from keycloak api to make sure